components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -795,6 +813,35 @@ class SitePermissionsFeature(
         )
     }
 
+    @Suppress("LongParameterList")
+    @VisibleForTesting
+    internal fun createContentStorageAccessPermissionPrompt(
+        context: Context,
+        host: String,
+        permissionRequest: PermissionRequest,
+        @StringRes titleId: Int,
+        @DrawableRes iconId: Int,
+        showDoNotAskAgainCheckBox: Boolean,
+        shouldSelectRememberChoice: Boolean,
+        isNotificationRequest: Boolean = false
+    ): SitePermissionsDialogFragment {
+        val currentSession = store.state.findTabOrCustomTabOrSelectedTab(sessionId)
+            ?: throw IllegalStateException("Unable to find session for $sessionId or selected session")
+
+        val title = context.getString(titleId, host, currentSession.content.url)
```

Comment:
Desktop uses smaller URLs which are easier to read and reason about:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/11428869/145867175-b8f797b1-8265-42b0-9862-136de5e7217c.png">
vs
<img width="482" alt="image" src="https://user-images.githubusercontent.com/11428869/145867367-fde1e460-f24c-481e-b6cb-9a73b5cdeafd.png">

We could decide to use a similar approach for all prompts.

Reply:
Agree maybe [stripDefaultPort](https://github.com/mozilla-mobile/android-components/blob/3b2d7a30fe7a558bd6661c3152d5a611f88cac9d/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt#L171) or [getOrigin()](https://github.com/mozilla-mobile/android-components/blob/3b2d7a30fe7a558bd6661c3152d5a611f88cac9d/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt#L157) could help.

Reply:
@Mugurell what's the current copy we have on mobile for this dialog? Is it the short one "Allow .... to use cookies on ..." or is it the slightly longer on "Allow .... to use its cookies on ...? You may want to block access if it's not clear why.... "? Thanks!

Reply:
@topotropic Tested just now. We are using the slightly longer version with [these two strings](https://github.com/mozilla-mobile/android-components/blob/8107b87e0898f99baa900217d5cd8e214cef3ec0/components/feature/sitepermissions/src/main/res/values/strings.xml#L41-L46).

<img src="https://user-images.githubusercontent.com/11428869/173838673-bec11058-0746-4a03-b75c-eb9ce5de8600.png" width="50%" />


&nbsp;

Seems like desktop does the same:
<img width="430" alt="image" src="https://user-images.githubusercontent.com/11428869/173839800-b7e9e756-0e33-4903-9f1d-b73b16f422c0.png">

----
Will update our negative button to also say `Block` instead of `Don't allow` based on https://mozilla-hub.atlassian.net/browse/FNXV2-20626